README
NAME
    Task::BeLike::JONASBN - Collection of modules often used by jonasbn

SYNOPSIS
        perl -MCPAN -e 'install Task::BeLike::JONASBN'

CONTENTS
    *   App::Ack

    *   App::Prove

    *   AppConfig

    *   CPAN::Mini

    *   CPAN::Reporter

    *   Class::Accessor

    *   Cwd

    *   DBD::Mock

    *   DBI

    *   Data::Dumper

    *   Data::FormValidator

    *   Date::Calc

    *   DateTime

    *   Devel::Cover

    *   Error

    *   ExtUtils::Install, 1.47

    *   File::Copy

    *   File::Find::Rule

    *   File::Path

    *   File::Slurp

    *   File::Which

    *   Hash::Merge

    *   JSON

    *   List::MoreUtils

    *   Module::Build, 0.35

    *   Module::Info::File

    *   Module::Load

    *   Params::Validate, 0.85

    *   Perl::Critic

    *   Perl::Critic

    *   Perl::Critic::Bangs

    *   Perl::Tidy

    *   Pod::Simple::Wiki

    *   Spreadsheet::WriteExcel

    *   TAP::Formatter::JUnit

    *   Task::Jenkins

    *   Template

    *   Term::ReadLine::Perl

    *   Test::Class

    *   Test::DatabaseRow

    *   Test::Exception

    *   Test::Fatal

    *   Test::Kwalitee

    *   Test::MockObject::Extends

    *   Test::More

    *   Test::Perl::Critic

    *   Test::Pod, 1.14

    *   Test::Pod::Coverage, 1.04

    *   Test::Prereq

    *   Tie::IxHash

    *   Try::Tiny

    *   WWW::Mechanize

    *   WWW::Mechanize::Cached

    *   XML::Simple

    *   YAML

    *   YAML::Shell

    *   local::lib

COPYRIGHT & LICENSE
    Copyright 2009-2014 Jonas B. Nielsen (jonasbn), All Rights Reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
Changes file for Task::BeLike::JONASBN

$Id$

Project site changelog:

http://logicLAB.jira.com/browse/BJONASBN#selectedTab=com.atlassian.jira.plugin.system.project%3Achangelog-panel

1.07 

- Update not required

- Enabled warnings pragma


1.06 2014-01-18

- Update not required

- Changed repository to Github

- Added Perl::Critic::logicLAB

- Added Task::Jenkins

- Specified requirement for Devel::Cover to be atleast version 1.00 [BJONASBN-54]

  https://logiclab.jira.com/browse/BJONASBN-54

- Added Test::CPAN::Changes to list of distributions I like [BJONASBN-53]

  https://logiclab.jira.com/browse/BJONASBN-53

- Added Test::CPAN::Changes test (t/changes.t), implementing automatic integrity test of 
  the Changes file, see also BJONASBN-50 [BJONASBN-52]

  https://logiclab.jira.com/browse/BJONASBN-52

- Added Test::Kwalitee test (t/kwalitee.t) [BJONASBN-55]

  https://logiclab.jira.com/browse/BJONASBN-55
  

1.05 2013-08-01

- Fixed up Changes file as part of my Questhub quest adhering to the standard
  described in: CPAN::Changes::Spec [BJONASBN-50]

  Quest: http://questhub.io/realm/perl/quest/51f0fd8e18ba7dcf4b000040
  Spec: https://metacpan.org/module/CPAN::Changes::Spec
  
  https://logiclab.jira.com/browse/BJONASBN-50


1.04 2012-12-03

- Maintenance release, update not required

- [BJONASBN-43] Added File::Find::Rule
  http://logiclab.jira.com/browse/BJONASBN-43

- [BJONASBN-48] Added File::Which
  http://logiclab.jira.com/browse/BJONASBN-48

- [BJONASBN-47] Added Test::DatabaseRow
  http://logiclab.jira.com/browse/BJONASBN-47

- [BJONASBN-41] Added Tie::IxHash
  http://logiclab.jira.com/browse/BJONASBN-41

- [BJONASBN-44] Added Perl::Critic::Bangs
  http://logiclab.jira.com/browse/BJONASBN-44

- [BJONASBN-49] Added YAML::Shell
  http://logiclab.jira.com/browse/BJONASBN-49

- [BJONASBN-46] Added local::lib
  http://logiclab.jira.com/browse/BJONASBN-46

- Changed timestamps in Changes file to ISO8601 format (my personal favorite)


1.03 2011-04-11

- Maintenance release, update not required

- [BJONASBN-38] Added App::Prove
  http://logiclab.jira.com/browse/BJONASBN-38

- [BJONASBN-36] Added Perl::Critic
  http://logiclab.jira.com/browse/BJONASBN-36

- [BJONASBN-35] Added TAP::Formatter::JUnit
  http://logiclab.jira.com/browse/BJONASBN-35

- [BJONASBN-37] Added Test::Fatal
  http://logiclab.jira.com/browse/BJONASBN-37

- [BJONASBN-34] Added Try::Tiny
  http://logiclab.jira.com/browse/BJONASBN-34


1.02 2010-11-24

- Maintenance release, update not required

- [BJONASBN-31]	Added App::Ack
  http://logiclab.jira.com/browse/BJONASBN-31

- [BJONASBN-32]	Add File::Copy
  http://logiclab.jira.com/browse/BJONASBN-32

- [BJONASBN-29]	Extracted and released Module::Build::Bundle in a bundle
  compatible version separate distribution
  http://logiclab.jira.com/browse/BJONASBN-29


1.01 2010-09-04

- Bugfix release, update not required

- [BJONASBN-25] addressed failing test. Adhering to documentation from
  Module::Build on requirements needed for configuration


1.00 2010-09-01

- Initial release, new packaging

- [BJONASBN-21] readdressed, POD now nicely formatted again

- Renamed distribution to Task::BeLike::JONASBN [BJONASBN-22]
  Bundle::JONASBN, will eventually be removed from CPAN, but I will leave it
  there for some time. I will not support it any longer. Task::BeLike::JONASBN
  will be the new distribution

- Implemented respect for build action, so the contents section in the 
  Perl module is auto-generated based on the custom build module
  Module::Build::Bundle [BJONASBN-16]

################################################
#                                              #
# Under the distribution name: Bundle::JONASBN #
#                                              #
################################################

0.04 2010-01-30

- Update required

- Added POD syntax test to test suite

- Addressed failing tests, my perception of a bundle does not seem to be a
real bundle


0.03 2009-10-08

- Update not required

- Added DataFormValidator [BJONASBN-13]

- Added CPAN::Mini [BJONASBN-14] 

- Added CPAN::Reporter [BJONASBN-15] 

- Added Pod::Simple::Wiki [BJONASBN-11] 

- Added DBD:.Mock [BJONASBN-9] 

- Added DateTime [BJONASBN-10] 


0.02 2009-10-05

- Update not required

- Added sorting to Build contents action [BJONASBN-7]

- Added WWW::Mechanize [BJONASBN-2]

- Added WWW::Mechanize::Cached [BJONASBN-3]

- Added Test::MockObject::Extends [BJONASBN-4]

- Added File::Slurp [BJONASBN-5]

- Added Devel::Cover [BJONASBN-6]

- Added Changes file [BJONASBN-8] 


0.01 2009-09-28

- Initial release


lib/Task/BeLike/JONASBN.pm
package Task::BeLike::JONASBN;

# $Id$

# Courtesy of Adam Kennedy
# http://search.cpan.org/dist/Task/lib/Task.pm

use strict;
use warnings;

use 5.006; # 5.6.0

our $VERSION = '1.06';

1;

__END__

=pod

=head1 NAME

Task::BeLike::JONASBN - Collection of modules often used by jonasbn

=head1 SYNOPSIS

    perl -MCPAN -e 'install Task::BeLike::JONASBN'

=head1 CONTENTS

=over

=item * L<App::Ack|App::Ack>

=item * L<App::Prove|App::Prove>

=item * L<AppConfig|AppConfig>

=item * L<CPAN::Mini|CPAN::Mini>

=item * L<CPAN::Reporter|CPAN::Reporter>

=item * L<Class::Accessor|Class::Accessor>

=item * L<Cwd|Cwd>

=item * L<DBD::Mock|DBD::Mock>

=item * L<DBI|DBI>

=item * L<Data::Dumper|Data::Dumper>

=item * L<Data::FormValidator|Data::FormValidator>

=item * L<Date::Calc|Date::Calc>

=item * L<DateTime|DateTime>

=item * L<Devel::Cover|Devel::Cover>

=item * L<Error|Error>

=item * L<ExtUtils::Install|ExtUtils::Install>, 1.47

=item * L<File::Copy|File::Copy>

=item * L<File::Find::Rule|File::Find::Rule>

=item * L<File::Path|File::Path>

=item * L<File::Slurp|File::Slurp>

=item * L<File::Which|File::Which>

=item * L<Hash::Merge|Hash::Merge>

=item * L<JSON|JSON>

=item * L<List::MoreUtils|List::MoreUtils>

=item * L<Module::Build|Module::Build>, 0.35

=item * L<Module::Info::File|Module::Info::File>

=item * L<Module::Load|Module::Load>

=item * L<Params::Validate|Params::Validate>, 0.85

=item * L<Perl::Critic|Perl::Critic>

=item * L<Perl::Critic|Perl::logicLAB>

=item * L<Perl::Critic::Bangs|Perl::Critic::Bangs>

=item * L<Perl::Tidy|Perl::Tidy>

=item * L<Pod::Simple::Wiki|Pod::Simple::Wiki>

=item * L<Spreadsheet::WriteExcel|Spreadsheet::WriteExcel>

=item * L<TAP::Formatter::JUnit|TAP::Formatter::JUnit>

=item * L<Task::Jenkins|Task::Jenkins>

=item * L<Template|Template>

=item * L<Term::ReadLine::Perl|Term::ReadLine::Perl>

=item * L<Test::Class|Test::Class>

=item * L<Test::CPAN::Changes>

=item * L<Test::DatabaseRow|Test::DatabaseRow>

=item * L<Test::Exception|Test::Exception>

=item * L<Test::Fatal|Test::Fatal>

=item * L<Test::Kwalitee|Test::Kwalitee>

=item * L<Test::MockObject::Extends|Test::MockObject::Extends>

=item * L<Test::More|Test::More>

=item * L<Test::Perl::Critic|Test::Perl::Critic>

=item * L<Test::Pod|Test::Pod>, 1.14

=item * L<Test::Pod::Coverage|Test::Pod::Coverage>, 1.04

=item * L<Test::Prereq|Test::Prereq>

=item * L<Tie::IxHash|Tie::IxHash>

=item * L<Try::Tiny|Try::Tiny>

=item * L<WWW::Mechanize|WWW::Mechanize>

=item * L<WWW::Mechanize::Cached|WWW::Mechanize::Cached>

=item * L<XML::Simple|XML::Simple>

=item * L<YAML|YAML>

=item * L<YAML::Shell|YAML::Shell>

=item * L<local::lib|local::lib>

=back

=head1 COPYRIGHT & LICENSE

Copyright 2009-2014 Jonas B. Nielsen (jonasbn), All Rights Reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut


Build.PL
# Build.PL for Bundle-JONASBN

# $Id$

use strict;
use warnings;
use Module::Build::Bundle;

our $VERSION = '0.05';

my $build = Module::Build::Bundle->new(
    dist_author        => 'Jonas B. Nielsen (jonasbn), <jonasbn@cpan.org>',
    module_name        => 'Task::BeLike::JONASBN',
    license            => 'artistic',
    create_makefile_pl => 'traditional',
    create_readme      => 1,
    configure_requires => {
        'Tie::IxHash' => '0',    #used in dist
    },
    requires => {
        'AppConfig'                   => '0',
        'App::Ack'                    => '0',
        'App::Prove'                  => '0',
        'Class::Accessor'             => '0',
        'CPAN::Mini'                  => '0',
        'CPAN::Reporter'              => '0',
        'Cwd'                         => '0',
        'DBD::Mock'                   => '0',
        'DBI'                         => '0',
        'Data::Dumper'                => '0',
        'Data::FormValidator'         => '0',
        'Date::Calc'                  => '0',
        'DateTime'                    => '0',
        'Devel::Cover'                => '1.00',
        'Error'                       => '0',
        'ExtUtils::Install'           => '1.47',
        'File::Copy'                  => '0',
        'File::Find::Rule'            => '0',
        'File::Path'                  => '0',
        'File::Slurp'                 => '0',
        'File::Which'                 => '0',
        'Hash::Merge'                 => '0',
        'JSON'                        => '0',
        'List::MoreUtils'             => '0',
        'local::lib'                  => '0',
        'Module::Build'               => '0.35',
        'Module::Info::File'          => '0',
        'Module::Load'                => '0',
        'Params::Validate'            => '0.85',
        'Perl::Critic'                => '0',
        'Perl::Critic::logicLAB'      => '0',
        'Perl::Critic::Bangs'         => '0',
        'Perl::Tidy'                  => '0',
        'Pod::Simple::Wiki'           => '0',
        'Spreadsheet::WriteExcel'     => '0',
        'TAP::Formatter::JUnit'       => '0',
        'Task::Jenkins'               => '0',
        'Tie::IxHash'                 => '0',
        'Template'                    => '0',
        'Test::Class'                 => '0',
        'Test::CPAN::Changes'         => '0',
        'Test::DatabaseRow'           => '0',
        'Test::Exception'             => '0',
        'Test::Fatal'                 => '0',
        'Test::Kwalitee'              => '0',
        'Test::MockObject::Extends'   => '0',
        'Test::More'                  => '0',
        'Test::Perl::Critic'          => '0',
        'Test::Pod'                   => '1.14',
        'Test::Pod::Coverage'         => '1.04',
        'Test::Prereq'                => '0',
        'Term::ReadLine::Perl'        => '0',
        'Try::Tiny'                   => '0',
        'WWW::Mechanize'              => '0',
        'WWW::Mechanize::Cached'      => '0',
        'XML::Simple'                 => '0',
        'YAML'                        => '0',
        'YAML::Shell'                 => '0',
    },
    meta_merge => {
        resources => {
            homepage => 'http://logiclab.jira.com/wiki/display/BJONASBN/Home',
            bugtracker =>
                'http://rt.cpan.org/NoAuth/Bugs.html?Dist=Task-BeLike-JONASBN',
            repository => 'https://github.com/jonasbn/bjonasbn',
            ProjectChangelog =>
                'http://logiclab.jira.com/browse/BJONASBN#selectedTab=com.atlassian.jira.plugin.system.project%3Achangelog-panel',
            ProjectSite => 'http://logicLAB.jira.com/browse/BJONASBN',
            ProjectWiki =>
                'http://logiclab.jira.com/wiki/display/BJONASBN/Home',
        }
    },
);

$build->create_build_script();
